src/libutil/url.cc

```diff
@@ -448,15 +448,17 @@ static std::optional<ScpLike> parseScp(std::string_view s) noexcept
 
 ParsedURL fixGitURL(std::string_view url)
 {
-    try {
+    if (auto r = boost::urls::parse_uri(url); r && r->has_scheme() && r->has_authority()) {
         return parseURL(url);
-    } catch (BadURL &) {
     }
```

Comment:
It should really be a "try catch". If that doesn't work, we should perhaps change `parseURL`, or switch the order so we *first* do `parseScp` and *then* do `parseURL`.